fmn/backends/base.py

```diff
@@ -53,10 +53,11 @@ def base_url_with_trailing_slash(self) -> str:
     def api_url(self) -> str | None:
         return self.base_url
 
-    @ft_cache
     def __str__(self) -> str:
-        clsname = type(self).__name__
-        return f"{clsname}({self.base_url!r})"
+        if not hasattr(self, "_str"):
+            clsname = type(self).__name__
+            self._str = f"{clsname}({self.base_url!r})"
+        return self._str
```

Comment:
[I benchmarked it](https://gist.github.com/nphilipp/f211c8a016a8057e7f4b89f6b5530714) and try/except is about 30% faster than manual checks (`foo` is try/except, the others are variants of conditionals):

```nils@makake:~/test/python> ./try_except_vs_cond.py 
foo: 0.0718677620170638
bar: 0.10609414198552258
baz: 0.10942566799349152
nils@makake:~/test/python>
```

Reply:
Interesting! I was just trying to avoid the duplicate `return` calls. I'll go back to your version, thanks!

Reply:
Heh, talked about this with a friend and he pointed me to a flaw in how I benchmarked this: It assumes long-lived objects, but we don’t have that with `gen_requester()` and `get_last_messages()` from `fmn.api.handlers.utils`. For short-lived objects, the performance reverses: try/except is a bit more than twice as expensive as using conditionals (gist [here](https://gist.github.com/nphilipp/6c4400f97eac7258568e38fb939d84ab)):

```
nils@makake:~/test/python> ./try_except_vs_cond2.py 
foo: 0.8480359880195465
bar: 0.3848317340016365
baz: 0.3872268960112706
nils@makake:~/test/python>
```

I lean towards making these cache their `Requester` and `DatagrepperAsyncProxy` objects, respectively. What do you think?

Reply:
Hmm yeah I suppose they can live for the duration of the application.